vortexor/src/cli.rs

```diff
@@ -63,155 +33,84 @@ fn port_range_validator(port_range: &str) -> Result<(u16, u16), String> {
-                .num_args(1)
-                .default_value(max_connections_per_ipaddr_per_min_static)
-                .value_parser(clap::value_parser!(u64))
-                .help("Controls the rate of the clients connections per IpAddr per minute."),
-        )
-        .arg(
-            Arg::new("num_quic_endpoints")
-                .long("num-quic-endpoints")
-                .num_args(1)
-                .default_value(num_quic_endpoints_static)
-                .value_parser(clap::value_parser!(u64))
-                .help("The number of QUIC endpoints used for TPU and TPU-Forward. It can be increased to \
-                       increase network ingest throughput, at the expense of higher CPU and general \
-                       validator load."),
-        )
-        .arg(
-            Arg::new("max_streams_per_ms")
-                .long("max-streams-per-ms")
-                .num_args(1)
-                .default_value(max_streams_per_ms_static)
-                .value_parser(clap::value_parser!(u64))
-                .help("Max streams per second for a streamer."),
-        )
-        .arg(
-            Arg::new("tpu_coalesce_ms")
-                .long("tpu-coalesce-ms")
-                .value_name("MILLISECS")
-                .num_args(1)
-                .value_parser(clap::value_parser!(u64))
-                .help("Milliseconds to wait in the TPU receiver for packet coalescing."),
-        )
-        .arg(
-            Arg::new("logfile")
-                .long("log")
-                .value_name("FILE")
-                .num_args(1)
-                .help(
-                    "Redirect logging to the specified file, '-' for standard error. Sending the \
-                     SIGUSR1 signal to the vortexor process will cause it to re-open the log file.",
-                ),
-        )
-        .arg(
-            Arg::new("destination")
-                .long("destination")
-                .value_name("HOST:PORT")
-                .action(ArgAction::Append)
-                .num_args(1)
-                .value_parser(solana_net_utils::parse_host_port)
-                .help("The destination validator address to which the vortexor will forward transactions."),
-        )
+    version_static
 }
+
+fn get_default_port_range() -> &'static str {
+    let range = format!("{}-{}", VALIDATOR_PORT_RANGE.0, VALIDATOR_PORT_RANGE.1);
+    let range: &'static str = Box::leak(range.into_boxed_str());
+    range
+}
+
+fn get_default_tpu_coalesce_ms() -> &'static str {
+    let coalesce = DEFAULT_TPU_COALESCE.as_millis().to_string();
+    let coalesce: &'static str = Box::leak(coalesce.into_boxed_str());
+    coalesce
+}
+
+#[derive(Parser)]
+#[command(name=crate_name!(),version=get_version(), about=crate_description!(),
+    long_about = None, color=ColorChoice::Auto)]
+pub struct Cli {
+    /// Vortexor identity keypair
+    #[arg(long, num_args=1, value_parser=clap::value_parser!(PathBuf), required=true, value_name="KEYPAIR")]
```

Comment:
Do you have to specify value_parser? I think it will use appropriate one by default here. This works fine:
```
#[arg(short, long)]
        /// Directory with pcap files to write. Existing ontents will be destroyed!
        output: PathBuf,
```
num_args is also not necessary here. required=true is implicit unless the arg value is an Option<T>

Reply:
Done